src/transactionOnNetwork.ts

```diff
@@ -50,7 +50,7 @@ export class TransactionOnNetwork {
     data: Buffer = Buffer.from([]);
     version: number = 0;
     options: number = 0;
-    signature: string = "";
+    signature?: Uint8Array;
```

Comment:
Should not be optional and you can set the default value to an empty Uint8Array.